File: mobile-app/src/api/workflow.js
import request from '@/utils/request'

/**
 * 获取待审批任务列表
 * @param {Object} params - 查询参数
 * @returns {Promise} 返回待审批任务列表
 */
export function getPendingTasks(params) {
  return request({
    url: '/workflow-tasks/pending',
    method: 'GET',
    params
  })
}

/**
 * 审批通过任务
 * @param {number} id - 任务ID
 * @param {Object} data - 审批数据
 * @returns {Promise} 返回审批结果
 */
export function approveTask(id, data) {
  return request({
    url: `/workflow-tasks/${id}/approve`,
    method: 'POST',
    data
  })
}

/**
 * 审批拒绝任务
 * @param {number} id - 任务ID
 * @param {Object} data - 拒绝原因
 * @returns {Promise} 返回拒绝结果
 */
export function rejectTask(id, data) {
  return request({
    url: `/workflow-tasks/${id}/reject`,
    method: 'POST',
    data
  })
}

/**
 * 任务退回
 * @param {number} id - 任务ID
 * @param {Object} data - 退回原因
 * @returns {Promise} 返回退回结果
 */
export function returnTask(id, data) {
  return request({
    url: `/workflow-tasks/${id}/return`,
    method: 'POST',
    data
  })
}

/**
 * 任务评论
 * @param {number} id - 任务ID
 * @param {Object} data - 评论内容
 * @returns {Promise} 返回评论结果
 */
export function commentTask(id, data) {
  return request({
    url: `/workflow-tasks/${id}/comment`,
    method: 'POST',
    data
  })
}

/**
 * 获取工作流定义列表
 * @returns {Promise} 返回工作流定义列表
 */
export function getWorkflows() {
  return request({
    url: '/workflows',
    method: 'GET'
  })
}

/**
 * 获取工作流实例列表
 * @param {Object} params - 查询参数
 * @returns {Promise} 返回工作流实例列表
 */
export function getWorkflowInstances(params) {
  return request({
    url: '/workflow-instances',
    method: 'GET',
    params
  })
}

/**
 * 获取工作流实例详情
 * @param {number} id - 实例ID
 * @returns {Promise} 返回实例详情
 */
export function getWorkflowInstance(id) {
  return request({
    url: `/workflow-instances/${id}`,
    method: 'GET'
  })
}

/**
 * 获取实例审批历史
 * @param {number} id - 实例ID
 * @returns {Promise} 返回审批历史
 */
export function getInstanceApprovals(id) {
  return request({
    url: `/workflow-instances/${id}/approvals`,
    method: 'GET'
  })
}

/**
 * 重新提交实例
 * @param {number} id - 实例ID
 * @returns {Promise} 返回提交结果
 */
export function resubmitInstance(id) {
  return request({
    url: `/workflow-instances/${id}/resubmit`,
    method: 'POST'
  })
}

File: mcp-server/tools/workflow.py
"""
Workflow Tools for MCP Server
Provides tools for workflow operations and approvals
"""

from typing import Any
import httpx


async def get_pending_tasks(
    backend_url: str,
    headers: dict,
    limit: int = 20,
) -> dict[str, Any]:
    """Get pending workflow tasks

    Args:
        backend_url: Backend API base URL
        headers: Authentication headers
        limit: Maximum number of results

    Returns:
        Pending tasks data
    """
    async with httpx.AsyncClient() as client:
        response = await client.post(
            f"{backend_url}/api/agent/operate",
            headers=headers,
            json={
                "operation": "query",
                "resource": "workflow",
                "parameters": {"status": "pending", "limit": limit},
                "reasoning": "Get pending workflow tasks",
            },
        )
        response.raise_for_status()
        return response.json()


async def approve_workflow(
    backend_url: str,
    headers: dict,
    task_id: int,
    remark: str = "",
) -> dict[str, Any]:
    """Approve a workflow task

    Args:
        backend_url: Backend API base URL
        headers: Authentication headers
        task_id: Workflow task ID
        remark: Approval remarks

    Returns:
        Approval result
    """
    async with httpx.AsyncClient() as client:
        response = await client.post(
            f"{backend_url}/api/agent/workflow",
            headers=headers,
            json={
                "task_id": task_id,
                "action": "approve",
                "remark": remark,
            },
        )
        response.raise_for_status()
        return response.json()


async def reject_workflow(
    backend_url: str,
    headers: dict,
    task_id: int,
    reason: str = "",
) -> dict[str, Any]:
    """Reject a workflow task

    Args:
        backend_url: Backend API base URL
        headers: Authentication headers
        task_id: Workflow task ID
        reason: Rejection reason

    Returns:
        Rejection result
    """
    async with httpx.AsyncClient() as client:
        response = await client.post(
            f"{backend_url}/api/agent/workflow",
            headers=headers,
            json={
                "task_id": task_id,
                "action": "reject",
                "remark": reason,
            },
        )
        response.raise_for_status()
        return response.json()


async def get_workflow_history(
    backend_url: str,
    headers: dict,
    instance_id: int,
) -> dict[str, Any]:
    """Get workflow approval history

    Args:
        backend_url: Backend API base URL
        headers: Authentication headers
        instance_id: Workflow instance ID

    Returns:
        Workflow history
    """
    async with httpx.AsyncClient() as client:
        response = await client.get(
            f"{backend_url}/api/workflow-instances/{instance_id}/approvals",
            headers=headers,
        )
        response.raise_for_status()
        return response.json()
